lib/ai_client.rb

```diff
@@ -0,0 +1,72 @@
+require "openai"
+
+class AiClient
+  DIFF_LIMIT = 10000
+
+  attr_reader :access_token, :client
+
+  def initialize(access_token = nil)
+    @access_token = access_token || Envs::OpenaiApiToken.new.fetch!
+    @client = OpenAI::Client.new(access_token: @access_token)
+  end
+
+  def chat(diff)
+    messages = set_messages(diff)
+    response = safe_api_request do
+      @client.chat(
+        parameters: {
+          model: selected_model,
+          messages: messages
+        }
+      )
+    end
+
+    response.dig("choices", 0, "message", "content")&.strip&.delete("\"")
+  end
+
+  def models_list
+    safe_api_request { @client.models.list }
+  end
+
+  def verify_api_token!
+    models_list
+  end
+
+  private
+
+  def safe_api_request
```

Comment:
move the error handling about `OpenAI::Client` here.
*was inside both `CommitMessageGenerator` and `Aicommit`